### ISSUE TITLE
eIDAS Node doesn't propagate relay state in light response

### ISSUE THREAD

**Issue by fenryxo** [CZ-NIC/django-eidas-specific-node#22]
**Upstream ticket:** EID-977 in CEF issue tracker.

Our light response stored by Specific Proxy Service:

```xml
<?xml version='1.0' encoding='utf-8' standalone='yes'?>
<lightResponse>
  <id>T94492c61-c8a8-4a6a-aee9-6ef6b02f213a</id>
  <inResponseToId>_cUZBLMtWjycfF4XeCy-nWYQXfJ98N0sDFuAlQzrFRDWTi1laUtkPIWlwuEtQccX</inResponseToId>
  <issuer>specific-proxy-service</issuer>
  <ipAddress>217.31.205.1</ipAddress>
  <relayState>xxxx</relayState>
  <subject>e9ef6043-99f6-4a6d-bf7b-956b560f6550</subject>
  <subjectNameIdFormat>urn:oasis:names:tc:SAML:2.0:nameid-format:persistent</subjectNameIdFormat>
  <levelOfAssurance>http://eidas.europa.eu/LoA/substantial</levelOfAssurance>
  <status>
    <failure>false</failure>
    <statusCode>urn:oasis:names:tc:SAML:2.0:status:Success</statusCode>
  </status>
  <attributes>
    <attribute>
      <definition>http://eidas.europa.eu/attributes/naturalperson/CurrentFamilyName</definition>
      <value>DVOŘÁKOVÁ</value>
    </attribute>
    <attribute>
      <definition>http://eidas.europa.eu/attributes/naturalperson/CurrentGivenName</definition>
      <value>PAVLA</value>
    </attribute>
    <attribute>
      <definition>http://eidas.europa.eu/attributes/naturalperson/DateOfBirth</definition>
      <value>1955-06-07</value>
    </attribute>
    <attribute>
      <definition>http://eidas.europa.eu/attributes/naturalperson/PersonIdentifier</definition>
      <value>e9ef6043-99f6-4a6d-bf7b-956b560f6550</value>
    </attribute>
  </attributes>
</lightResponse>
```

Light response provided by Generic Connector:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<lightResponse>
    <id>_c.fR2g8Vq1miFuxjORQlvumxjzU4XQqbqNPn-Z-gRTZegbXmaM2qRbGZmYLfMwK</id>
    <issuer>http://pokuston:8888/EidasNode/ConnectorMetadata</issuer>
    <ipAddress>2001:1488:fffe:2:b86f:bcfd:5dc7:8228</ipAddress>
    <subject>e9ef6043-99f6-4a6d-bf7b-956b560f6550</subject>
    <subjectNameIdFormat>urn:oasis:names:tc:SAML:2.0:nameid-format:persistent</subjectNameIdFormat>
    <status>
        <failure>false</failure>
        <statusCode>urn:oasis:names:tc:SAML:2.0:status:Success</statusCode>
        <statusMessage>urn:oasis:names:tc:SAML:2.0:status:Success</statusMessage>
        <subStatusCode>##</subStatusCode>
    </status>
    <inResponseToId>_16f32676-1c31-44d8-a2fe-3a100cf29c0e</inResponseToId>
    <levelOfAssurance>http://eidas.europa.eu/LoA/substantial</levelOfAssurance>
    <attributes>
        <attribute>
            <definition>http://eidas.europa.eu/attributes/naturalperson/CurrentFamilyName</definition>
            <value>DVOŘÁKOVÁ</value>
        </attribute>
        <attribute>
            <definition>http://eidas.europa.eu/attributes/naturalperson/CurrentGivenName</definition>
            <value>PAVLA</value>
        </attribute>
        <attribute>
            <definition>http://eidas.europa.eu/attributes/naturalperson/DateOfBirth</definition>
            <value>1955-06-07</value>
        </attribute>
        <attribute>
            <definition>http://eidas.europa.eu/attributes/naturalperson/PersonIdentifier</definition>
            <value>CA/CA/e9ef6043-99f6-4a6d-bf7b-956b560f6550</value>
        </attribute>
    </attributes>
</lightResponse>
```

`<relayState>xxxx</relayState>` is missing in the latter.

**Comment by fenryxo** [comment]
The issue is reproducible also with Demo Identity Provider from eIDAS Node:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<lightResponse>
    <id>_zq4uVVyEKdDA6fpiAshencLy0ThBUcdHkSZtvAw-Hiz4ixIh36TqCeJDkJfWg2D</id>
    <issuer>http://pokuston:8888/EidasNode/ConnectorMetadata</issuer>
    <ipAddress>2001:1488:fffe:2:b86f:bcfd:5dc7:8228</ipAddress>
    <subject>0123456</subject>
    <subjectNameIdFormat>urn:oasis:names:tc:SAML:2.0:nameid-format:persistent</subjectNameIdFormat>
    <status>
        <failure>false</failure>
        <statusCode>urn:oasis:names:tc:SAML:2.0:status:Success</statusCode>
        <statusMessage>urn:oasis:names:tc:SAML:2.0:status:Success</statusMessage>
        <subStatusCode>##</subStatusCode>
    </status>
    <inResponseToId>_b26e18ff-7370-4abd-bce7-b26ba43a76dd</inResponseToId>
    <levelOfAssurance>http://eidas.europa.eu/LoA/low</levelOfAssurance>
    <attributes>
        <attribute>
            <definition>http://eidas.europa.eu/attributes/naturalperson/CurrentGivenName</definition>
            <value>javier</value>
        </attribute>
        <attribute>
            <definition>http://eidas.europa.eu/attributes/naturalperson/PersonIdentifier</definition>
            <value>CA/CA/12345</value>
        </attribute>
        <attribute>
            <definition>http://eidas.europa.eu/attributes/naturalperson/DateOfBirth</definition>
            <value>1964-12-31</value>
        </attribute>
        <attribute>
            <definition>http://eidas.europa.eu/attributes/naturalperson/CurrentFamilyName</definition>
            <value>Garcia</value>
        </attribute>
    </attributes>
</lightResponse>
```

**Comment by fenryxo** [comment]
`/EidasNode/SpecificProxyServiceResponse` of Generic Proxy Service sends `RelayState` in HTTP POST to `/EidasNode/ColleagueResponse` of Generic Connector, which should store it in light response.

**Comment by fenryxo** [comment]
`/EidasNode/ColleagueResponse` gets two HTTP POST parameters:

* `SAMLResponse` (here base64 decoded and simplified):

```xml
<saml2p:Response xmlns:ds="http://www.w3.org/2000/09/xmldsig#" xmlns:eidas="http://eidas.europa.eu/attributes/naturalperson" xmlns:saml2="urn:oasis:names:tc:SAML:2.0:assertion" xmlns:saml2p="urn:oasis:names:tc:SAML:2.0:protocol" Consent="urn:oasis:names:tc:SAML:2.0:consent:obtained" Destination="http://pokuston:8888/EidasNode/ColleagueResponse" ID="_d-AqgH432Lo-ep3ActuXrdjAD6IuNh8VtSATGbUCSzq9WjE16YZq9Npcc3W7Zp3" InResponseTo="_wf2YCsr3E6L0hJwG2ZhCMPY3O5wRcO0DqId2mje1Qf3TYaEuSKnDLL6tviGJu1o" IssueInstant="2019-09-06T12:24:08.835Z" Version="2.0">
<saml2:Issuer Format="urn:oasis:names:tc:SAML:2.0:nameid-format:entity">http://pokuston:8888/EidasNode/ServiceMetadata</saml2:Issuer>
<ds:Signature>...</ds:Signature>
<saml2p:Status>
<saml2p:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Success"/>
<saml2p:StatusMessage>urn:oasis:names:tc:SAML:2.0:status:Success</saml2p:StatusMessage>
</saml2p:Status>
<saml2:EncryptedAssertion>...</saml2:EncryptedAssertion>
</saml2p:Response>
```

* RelayState: `xxxx`

**Comment by jtalir** [comment]
This has been confirmed byt CEF team and will be fixed. It can be tracked as EID-977 in CEF issue tracker.

**Comment by fenryxo** [comment]
Fixed in eIDAS Node 2.4.
